src/cmd.rs

```diff
@@ -324,6 +324,20 @@ fn shorten_to(s: String, max_len: usize) -> String {
     }
 }
 
+/// Replace all characters that are not allowed as part of a pod name
+/// Regex for validation is:
+/// '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*'
+fn ensure_pod_name(s: String) -> String {
+    let pod_name_regex = Regex::new(r"[^a-z0-9-]").unwrap();
+    let result = pod_name_regex.replace_all(&s.to_ascii_lowercase().to_string(), "-").to_string();
+
+    if &result[0..1] == "-" || &result[58..59] == "-" {
```

Comment:
You can't hard-code up to 59 here since the pod name might be less than 59 characters which will cause a panic.  You'll need to look at the string length and go off that.

Reply:
Yeah, that's true.  The other option is that you can delete the pod when it's done.  Maybe the right thing to do would be to default to adding a random string, but have an option like `--name` that lets you set the name you want to use

Reply:
Good point!